### ISSUE TITLE
split romaji with whitespace

### ISSUE THREAD

**Issue by bitinn** [lovell/hepburn#12]
I understand it is not phonetically correct to insert whitespace between every kana, BUT, given I simply use `hepburn` as tokenizer for speed up search query (hence not using `kuroshiro` analyzer), is it possible to have an option to split result from `fromKana()`?

**Comment by bitinn** [comment]
My initial thought was to split before input but it didn't quite work due to `word.split('')` breaking romaji like `shī`.

```javascript
      let words = input.trim().split(' ');
      words.forEach(s => {
        if (s.length < 1) {
          return;
        }

        if (!hepburn.containsKana(s)) {
          searchWords.push(s);
          return;
        }

        let chars = s.split('');
        let temp = '';
        chars.forEach(c => {
          if (!hepburn.containsKana(c)) {
            temp += c;
            return;
          }

          if (temp != '') {
            searchWords.push(temp);
            temp = '';
          }

          searchWords.push(hepburn.fromKana(c).toLowerCase());
        });
```

**Comment by bitinn** [comment]
I am guessing my main problem is `ー` being split.

Which means a title like `はるかなレシーブ` will be `ha, ru, ka, na, re, shi, ー, bu` instead of `ha, ru, ka, na, re, shī, bu`. I am not sure if I can solve this without changing hepburn.

**Comment by lovell** [comment]
Hi David, this is a great question but before I continue you deserve a thank you for your `node-fetch` module that I've had the pleasure of using regularly over the last few years.

If the underlying problem is a lack of support for the choonpu symbol `ー` in JavaScript's `split` function, which is limited to UTF-16 codeunits, then perhaps a custom reducer might help:

```javascript
const choonpuReducer = (acc, cur) => {
  if (cur === "ー") {
    acc[acc.length - 1] += "ー";
  } else {
    acc.push(cur);
  }
  return acc;
};

const result = "はるかなレシーブ"
  .split("")
  .reduce(choonpuReducer , [])
  .map(hepburn.fromKana);
```
gives
```
["HA", "RU", "KA", "NA", "RE", "SHĪ", "BU"]
```

**Comment by bitinn** [comment]
@lovell thx for your kind words, I shouldn't take all the credit as v2 is largely managed by TimothyGu. also, thx for your solid work on `sharp`!

And your answer does solve most of my issue, but does it mean a proper reducer need all of these: `ゅ, ょ, ー, ュ, ョ, ャ, ィ, ゥ ...` 

ref: https://github.com/lovell/hepburn/blob/master/lib/hepburn.js#L56

Perhaps there should be a reference somewhere or a helper function

**Comment by lovell** [comment]
There are 21 combining forms in total, 12 in Hiragana and 9 in Katakana:

https://en.wiktionary.org/wiki/Category:Japanese_combining_forms_in_Hiragana_script
https://en.wiktionary.org/wiki/Category:Japanese_combining_forms_in_Katakana_script

How about a `splitKana()` function that converts a string of Kana into an array of combined chars ready for further processing?

```javascript
splitKana("はるかなレシーブ") // ["は", "る", "か", "な", "レ", "シー", "ブ"]
```

Happy to accept and/or help with a PR for this.

**Comment by bitinn** [comment]
Thx, I will look into it.

**Comment by lovell** [comment]
The `splitKana` and `splitRomaji` features are now available in v1.1.3.
